### ISSUE TITLE
[BUG] unordered_map_concurrent does not decrement size in erase

### ISSUE THREAD

**Issue by ElaraFX** [OpenImageIO/oiio#2623]
**Describe the bug**
In `unordered_map_concurrent::erase`, should add `--m_size` if an element is really erased.

**To Reproduce**
Check the code logic of `unordered_map_concurrent`

**Expected behavior**
The `unordered_map_concurrent::m_size` should reflect the actual size in container.

**Evidence**
```
unordered_map_concurrent<int, int> my_map;
int key=1;
int value=2;
my_map.insert(key, value);
my_map.erase(key);
std::cout << my_map.size() << std::endl; // should be 0 here!
```

**Platform information:**
 - OIIO branch/version: Release 1.8.17
 - OS: CentOS 7.5
 - C++ compiler: GCC 8.1
 - Any non-default build flags when you build OIIO:

**IF YOU ALREADY HAVE A CODE FIX:** There is no need to file a separate issue,
please just go straight to making a [pull request](https://github.com/OpenImageIO/oiio/pulls).

**Comment by lgritz** [comment]
You're quite right, I will submit a PR immediately to fix.

Unfortunately, I don't anticipate backporting to 1.8, it's too old. The current supported release where we routinely backport bug fixes is 2.1. But if you wish to maintain a copy on your side, it should be really simply to make the one-line fix on your end.

**Comment by lgritz** [comment]
Proposed patch #2624

**Comment by ElaraFX** [comment]
> You're quite right, I will submit a PR immediately to fix.
> 
> Unfortunately, I don't anticipate backporting to 1.8, it's too old. The current supported release where we routinely backport bug fixes is 2.1. But if you wish to maintain a copy on your side, it should be really simply to make the one-line fix on your end.

No problem. Thanks for the quick patch!
